### PR TITLE
Add BBEs for request and request error interceptors

### PR DESCRIPTION
**Pull request by TharmiganK**
## Purpose

Adding BBEs for request interceptors and request error interceptors with the recent changes. (Targeted for Beta 6)

> Fixes [`Update HTTP interceptor BBEs #2460`](https://github.com/ballerina-platform/ballerina-standard-library/issues/2460)

### CODE REVIEW

**Review by ayeshLK — APPROVED** [review]

File: examples/http-request-interceptor-at-service/http_request_interceptor_at_service.bal [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        http:Response res = new;
+        // Set the interceptor headers from request
+        res.setHeader(interceptor_header1, 
+                            check req.getHeader(interceptor_header1));
+        res.setHeader(interceptor_header2, 
+                            check req.getHeader(interceptor_header2));
+        res.setHeader(interceptor_header3, 
+                            check req.getHeader(interceptor_header3));
+        res.setTextPayload("Greetings!");
+        check caller->respond(res);
```
Comment (chamil321) [inline examples/http-request-interceptor-at-service/http_request_interceptor_at_service.bal:79]: shall we return response here rather using caller?

File: examples/http-request-interceptor-at-listener/http_request_interceptor_at_listener.bal [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            returns error? {
+        // Create a new response.
+        http:Response res = new;
+        // Set the interceptor headers from request
+        res.setHeader(interceptor_header1, 
+                            check req.getHeader(interceptor_header1));
+        res.setHeader(interceptor_header2, 
+                            check req.getHeader(interceptor_header2));
+        res.setTextPayload("Greetings!");
+        check caller->respond(res);
```
Comment (chamil321) [inline examples/http-request-interceptor-at-listener/http_request_interceptor_at_listener.bal:65]: Here also shall we return response?

File: examples/http-request-interceptor-at-service/http_request_interceptor_at_service.bal (hunk 1 above)
Comment (TharmiganK) [reply examples/http-request-interceptor-at-service/http_request_interceptor_at_service.bal]: Fixed in [`b2c2d40`](https://github.com/ballerina-platform/ballerina-distribution/pull/2399/commits/b2c2d40ca008262381cc5180f92b406076aef00f)

File: examples/http-request-interceptor-at-listener/http_request_interceptor_at_listener.bal (hunk 2 above)
Comment (TharmiganK) [reply examples/http-request-interceptor-at-listener/http_request_interceptor_at_listener.bal]: Fixed in [`b2c2d40`](https://github.com/ballerina-platform/ballerina-distribution/pull/2399/commits/b2c2d40ca008262381cc5180f92b406076aef00f)

**Review by chamil321 — APPROVED** [review]

**Pull request merged**